src/subscribables/observableUtils.js

```diff
@@ -1,15 +1,22 @@
 ko.when = function(predicate, callback, context) {
-    var observable = ko.pureComputed(predicate, context).extend({notify:'always'});
-    var subscription = observable.subscribe(function(value) {
-        if (value) {
-            subscription.dispose();
-            callback.call(context, value);
-        }
-    });
-    // In case the initial value is true, process it right away
-    observable['notifySubscribers'](observable.peek());
+    function kowhen (resolve) {
+        var observable = ko.pureComputed(predicate, context).extend({notify:'always'});
```

Comment:
For performance, would it make sense if this was always a deferred observable?

We may have to change the initial test so it resolves synchronously (unless `ko.when` is to be always async), but otherwise that *feels* like a good internal boost. :)